scripts/Diffraction/isis_powder/routines/focus.py

```diff
@@ -137,32 +142,46 @@ def _focus_one_ws(
         # Reduce the bin width if required on this instrument
         output_spectra = common.rebin_workspace_list(workspace_list=output_spectra, bin_width_list=bin_widths)
 
-    # Output
-    d_spacing_group, tof_group = instrument._output_focused_ws(output_spectra, run_details=run_details)
-
-    common.keep_single_ws_unit(d_spacing_group=d_spacing_group, tof_group=tof_group, unit_to_keep=instrument._get_unit_to_keep())
-
     # Tidy workspaces from Mantid
     common.remove_intermediate_workspace(input_workspace)
     common.remove_intermediate_workspace(aligned_ws)
     common.remove_intermediate_workspace(focused_ws)
-    common.remove_intermediate_workspace(output_spectra)
 
-    return d_spacing_group
+    return output_spectra
 
 
-def _apply_vanadium_corrections(instrument, input_workspace, perform_vanadium_norm, vanadium_splines):
+def _apply_vanadium_corrections(instrument, input_workspace, perform_vanadium_norm, vanadium_ws):
     input_workspace = mantid.ConvertUnits(InputWorkspace=input_workspace, OutputWorkspace=input_workspace, Target="TOF")
     split_data_spectra = common.extract_ws_spectra(input_workspace)
 
     if perform_vanadium_norm:
-        processed_spectra = _normalize_spectra(spectra_list=split_data_spectra, vanadium_splines=vanadium_splines, instrument=instrument)
+        processed_spectra = _normalize_spectra(spectra_list=split_data_spectra, vanadium_ws=vanadium_ws, instrument=instrument)
     else:
         processed_spectra = split_data_spectra
 
     return processed_spectra
 
 
+def _get_vanadium_ws(instrument, run_details):
+    van = "van_{}".format(run_details.vanadium_run_numbers)
+    if van in mantid.mtd:
+        return mantid.mtd[van]
+
+    if instrument.get_instrument_prefix() != "OSIRIS":
```

Comment:
Please can you define a function in abstract_inst.py called something like `get_vanadium_path` that returns `run_details.splined_vanadium_file_path`. You can then override this in osiris.py and return `run_details.unsplined_vanadium_file_path`. Finally you can call that function here and avoid the if instrument="OSIRIS" statement